Create ns_speex_tilde

michalsoloducha opened the pull request:
I have tested it for different input parameters and it seems to work fine. Feel free to add modifications/comments.

Commit:
Create ns_speex_tilde

I have tested it for different input parameters and it seems to work fine. Feel free to add modifications/comments.

Commit:
Update CMakeLists.txt

added ns_speex

dennisguse commented on src/signal-processing/ns_speex_tilde:
```diff
@@ -0,0 +1,167 @@
+#include <limits.h>
+#include <stdbool.h>
+#include <string.h>
+#include "ringbuffer.h"
```
Should not be required.
I will cleanup this include everywhere else.

dennisguse commented on src/signal-processing/ns_speex_tilde:
```diff
@@ -0,0 +1,167 @@
+
+} t_ns_speex_tilde;
+
+
```
Remove empty line

dennisguse commented on src/signal-processing/ns_speex_tilde:
```diff
@@ -0,0 +1,167 @@
+
+  x->speex_preprocess_state = speex_preprocess_state_init (x->codec.frame_size, x->codec.sample_rate_internal);
+
+  // enable noise suppression
```
Remove comment

dennisguse replied:
And also additional comments, e.g. constructor, destructure etc.

dennisguse commented on src/signal-processing/ns_speex_tilde:
```diff
@@ -0,0 +1,167 @@
+
+// setup new class
+void ns_speex_tilde_setup (void) {
+  ns_speex_tilde_class = class_new (
```
Please apply the gnu intend tool as described in the README.md

michalsoloducha replied:
I can't find that information in the main readme file.

dennisguse replied:
Ohh my fault - it is in the DEVELOPER.md

```
indent \
--braces-after-if-line \
--braces-after-func-def-line \
--braces-on-if-line \
--braces-on-func-def-line \
--braces-on-struct-decl-line \
--break-after-boolean-operator \
--break-before-boolean-operator \
--dont-break-function-decl-args \
--dont-break-function-decl-args-end \
--dont-break-procedure-type \
--dont-star-comments \
--space-after-cast \
--no-tabs \
--cuddle-else \
-l999 \
*.{c,h}
```

dennisguse commented on src/signal-processing/ns_speex_tilde:
```diff
@@ -0,0 +1,167 @@
+	(t_method) ns_speex_tilde_free, 
+	sizeof(t_ns_speex_tilde), 
+	CLASS_DEFAULT,
+	A_DEFFLOAT, A_DEFFLOAT, A_DEFFLOAT, 0 // init. values for: frame size, sample rate, max. noise attenuation
```
Remove comment

dennisguse commented on src/signal-processing/ns_speex_tilde:
```diff
@@ -0,0 +1,167 @@
+/**
+@file ns_speex_tilde.c
```
Can you rename it to denoise_speex_tilde.c
This allows to understand easier what it actually does/

dennisguse commented on src/signal-processing/ns_speex_tilde:
```diff
@@ -0,0 +1,167 @@
+  ns_speex~ FRAME_SIZE SAMPLE_RATE MAX_NOISE_ATT
+  FRAME_SIZE: 80, 160, 240, 320, 640 [samples]
+  SAMPLE_RATE: 8000, 16000, 32000 [Hz]
+  MAX_NOISE_ATT - maximum noise attenuation (negative number) [dB]
```
@MAX_NOISE_ATT

I would prefer the long version (MAX_NOISE_ATTENUATION) and also the expected range -100..0?
What is the default value?

dennisguse commented on src/signal-processing/ns_speex_tilde:
```diff
@@ -0,0 +1,167 @@
+typedef struct _ns_speex_tilde {
+  t_object x_obj;
+  
+  t_int max_noise_att;
```
long version

dennisguse commented:
Thanks Michal!
Its a good job!

I just added some comments, and when resolved I will merge it.
And could you add a help file?
I just saw that vad_speex~ also misses the help file and it would be awesome if you could also add one for this external.

dennisguse reviewed (changes requested):
Looks good - just requires some smaller changes.

Commit:
Merge remote-tracking branch 'refs/remotes/TheTelephone/master' into patch-1

Commit:
function name changed to denoise_speex_tilde

Commit:
Addressing requested changes

Commit:
Test commit

Commit:
GNU Indent tool applied

Commit:
Deleted some unnecessary links to header files

michalsoloducha commented:
Allright, all the requested changes are done. At some point later I will provide the help files for denoise and vad.

dennisguse committed:
Formatting

dennisguse commented:
Code looks good.
I would like to wait until the help file is also ready.
Then it is just one commit ;)

Commit:
Help files for denoise_speex and vad_speex

Commit:
Merge branch 'patch-1' of https://github.com/michalsoloducha/TheTelephone into patch-1

Commit:
dB -> dB

Commit:
Additional sample rate: 32000 Hz

michalsoloducha commented:
I have uploaded the help files for denoise_speex and vad_speex. Plus some minor changes.

dennisguse committed:
Fixed includes and added @see

dennisguse committed:
Update denoise_speex_tilde.c

dennisguse committed:
Update denoise_speex_tilde.c

dennisguse committed:
Update vad_speex_tilde.c

dennisguse committed:
Update denoise_speex_tilde.c

dennisguse committed:
Update denoise_speex_tilde.c

dennisguse commented:
One final thing: is t_outlet actually used by denoise_speex_tilde.c?
The required outlet should be made available by generic_codec.h, or?

Could you test if t_outlet is required, and if not remove it?

Commit:
Few minor changes:
- deleted unnecessary audio outlet declaration
- corrected error message in case of wrong max. noise attenuation

michalsoloducha commented:
Ok, so I got rid of unnecessary audio outlet declaration made some minor fixes.
